src/test/java/com/fittracker/fittracker/service/WeightServiceTest.java

```diff
@@ -185,5 +193,60 @@ void givenNoDateFound_shouldThrowException() {
             verifyNoMoreInteractions(weightRepository);
         }
     }
+    @Nested
+    class FindByDateRange {
+        private final LocalDate endDate = TEST_DATE.plusDays(3);
+        private final Weight endWeight = new Weight(endDate,TEST_VALUE,TEST_UUID);
+        @Test
+        void givenStartDateAfterEndDate_shouldThrowInvalidDateRangeException() {
+            assertThatThrownBy(()->weightService.findByDateRange(LocalDate.of(2023,5,1),LocalDate.of(2023,1,1)))
+                    .isInstanceOf(InvalidDateRangeException.class)
+                    .hasMessageContaining("Start date cannot be after end date");
+
+            verifyNoInteractions(weightRepository);
+        }
+
+        @Test
+        void givenValidDateRange_shouldReturnListOfWeightResponses() {
+            when(weightRepository.findByDateBetweenAndUserId(TEST_DATE, endDate,TEST_UUID))
+                    .thenReturn(List.of(weight,endWeight));
+
+            var expected = Stream.of(weight,endWeight).map(WeightResponse::fromWeight).toList();
```

Comment:
we're kind of copy pasting the implementation logic here, I'd rather see it like this:
```suggestion
var expected = List.of(new WeightResponse(TEST_DATE, TEST_VALUE), new WeightResponse(endDate, TEST_VALUE));
```